nsxt/resource_nsxt_manager_cluster.go

```diff
@@ -175,75 +132,58 @@ func getClusterInfoFromHostNode(d *schema.ResourceData, m interface{}) (string,
 	min := c.CommonConfig.MinRetryInterval
 	max := c.CommonConfig.MaxRetryInterval
 	maxRetries := c.CommonConfig.MaxRetries
-	hostIP := ""
+	hostIPs := []string{}
 	for i := 0; i < maxRetries; i++ {
 		clusterConfig, err := client.Get()
 		if err != nil {
-			return "", "", "", handleReadError(d, "Cluster Config", "", err)
+			return "", "", hostIPs, err
 		}
-		if hostIP == "" {
-			hostIP, err = getHostIPFromClusterConfig(m, clusterConfig)
+		if len(hostIPs) == 0 {
+			hostIPs, err = resolveHostIPs(m)
 			if err != nil {
-				return "", "", "", err
+				return "", "", hostIPs, err
 			}
+
+			log.Printf("[DEBUG]: Host resolved to IP addresses %v", hostIPs)
 		}
 		clusterID := *clusterConfig.ClusterId
 		nodes := clusterConfig.Nodes
 		node := nodes[0]
 		apiListenAddr := node.ApiListenAddr
 		if apiListenAddr != nil {
 			certSha256Thumbprint := *apiListenAddr.CertificateSha256Thumbprint
-			return clusterID, certSha256Thumbprint, hostIP, nil
+			return clusterID, certSha256Thumbprint, hostIPs, nil
 		}
 		interval := (rand.Intn(max-min) + min)
 		time.Sleep(time.Duration(interval) * time.Millisecond)
 		log.Printf("[DEBUG]: Waited %d ms before retrying getting API Listen Address, attempt %d", interval, i+1)
 	}
-	return "", "", "", fmt.Errorf("Failed to read ClusterConfig info from host node after %d attempts", maxRetries)
+	return "", "", hostIPs, fmt.Errorf("Failed to read ClusterConfig after %d attempts", maxRetries)
 }
 
-func getHostIPFromClusterConfig(client interface{}, clusterConfig nsxModel.ClusterConfig) (string, error) {
+func resolveHostIPs(client interface{}) ([]string, error) {
 	c := client.(nsxtClients)
 	host := c.Host
 	host = strings.TrimPrefix(host, "https://")
 	// Check if host is ip address or fqdn, if it's ip address then we are all set,
```

Comment:
with the approach in resolveHostIPs I think this comment section is no longer needed

Reply:
Updated the comment, thanks